examples/time_frequency/plot_compute_raw_data_spectrum.py

```diff
@@ -70,3 +71,18 @@
 
 ax.set_title('Four left-temporal magnetometers')
 plt.legend(['Without SSP', 'With SSP', 'SSP + Notch'])
+
+# Alternatively, you may also create PSDs from Raw objects with psd_XXX
+f, ax = plt.subplots()
+psds, freqs = psd_multitaper(raw, low_bias=True, tmin=tmin, tmax=tmax,
```

Comment:
> why do we allow filtering a subset of channels when the proj is for all
the channels?

There are legitimate use cases for this.  Imagine that exactly one EEG
channel had really bad line noise. You'd want to filter the line noise
(only from that channel is fine) before applying average ref.

Note that projectors are often calculated in a specific frequency band of
data, too -- look at our defaults for ECG and EOG, each one filters before
calculating proj. So often the filtering of raw data will differ from the
filtering used to calculate projectors. The filtering during proj is used
to better isolate the proper contaminating spatial component, but should
clean the unfiltered data fine.

> Can't you truncate proj when you filter with a subset

Projecting with fewer than the original channels used to determine the
basis vector for the projector, as was done in the PSD functions, should be
avoided if possible. Right now we don't have great safeguards against it
but we have recently discussed adding them.

> If I apply proj first, then filter (with picks), it is different from
filtering (with picks) and then applying proj?

It *can be* different. One operates spatially and the other temporally, but
both are linear operators. So the order does not matter if...

> there is no picks and you have all channels

...as, in that case, they are linear operations operating *on the same
data*,  so you can swap order without issue. But if you apply a linear
(temporal) operator to an *incomplete* subset of (spatial) channels, the
result will not be the same with the order switched, because the operators
are not both applied to the same data. Make sense?

Think about the bad 60 Hz contamination of a single EEG channel case I
mentioned above, and what happens if we notch the single channel before
applying proj, versus apply avg ref proj and notch it after, and hopefully
it will be clear.

Unfortunately I can't think of how to prevent users from making mistakes
from misunderstanding these issues without otherwise blocking legitimate
use cases...